packages/gateway-trpc/src/modules/offline/offline-router.ts

```diff
@@ -0,0 +1,30 @@
+import { PaginateInputSchema } from "@dotkomonline/core"
+import { OfflineSchema, OfflineWriteSchema } from "@dotkomonline/types"
+import { z } from "zod"
+import { protectedProcedure, t } from "../../trpc"
+
+export const offlineRouter = t.router({
+  create: t.procedure.input(OfflineWriteSchema).mutation(async ({ input, ctx }) => ctx.offlineService.create(input)),
+  edit: protectedProcedure
+    .input(
+      z.object({
+        id: OfflineSchema.shape.id,
+        input: OfflineWriteSchema.partial(),
+      })
+    )
+    .mutation(async ({ input: changes, ctx }) => ctx.offlineService.update(changes.id, changes.input)),
+  all: t.procedure
+    .input(PaginateInputSchema)
+    .query(async ({ input, ctx }) => ctx.offlineService.getAll(input.take, input.cursor)),
+  get: t.procedure.input(OfflineSchema.shape.id).query(async ({ input, ctx }) => ctx.offlineService.get(input)),
+  getS3UploadLink: protectedProcedure
```

Comment:
Maybe call this createUploadUrl? It being S3 is of insignificant meaning to the caller

Reply:
changed it to `createPresignedPost`, that's what it does